### PR TITLE
test(cli): cover profile state mutations

### LINKED ISSUE #422: Add coverage for profile CLI state mutations
> *This was generated by AI during triage.*

## Summary

The profile management CLI mutates auth/profile state, but coverage is thin around create/switch/delete/rename flows and their filesystem/config edge cases.

## Evidence

- `src/notebooklm/cli/profile.py:76` defines the profile command group.
- `src/notebooklm/cli/profile.py:135-290` implements create, switch, delete, and rename.
- Local coverage run reported `src/notebooklm/cli/profile.py` at 32% coverage.

## Proposed Fix

Add focused CLI tests for profile operations using isolated temp NotebookLM homes/config paths. Cover at least:

- invalid profile names
- creating an existing profile
- switching to a missing profile and showing available profiles
- writing `default_profile` with secure config permissions
- blocking deletion of active/default profiles
- deletion cancellation and `--confirm`
- rename success
- rename updates config when renaming the configured default
- config read/write failure paths where practical

## Acceptance Criteria

- Profile command tests cover mutating paths without touching the real user profile directory.
- Error messages and JSON/table outputs remain stable.
- Coverage for `src/notebooklm/cli/profile.py` is materially improved.

### PR DESCRIPTION
**Pull request by teng-lin**
## Summary
- add isolated CLI coverage for profile create, switch, delete, and rename state mutations
- factor profile config read/write helpers and preserve secure config permissions
- surface clean config write failures from `profile switch`

Fixes #422

## Test plan
- `uv run pytest tests/unit/cli/test_profile.py tests/unit/test_paths.py -q`
- `uv run pytest tests/unit/cli -q`
- `uv run ruff check src/notebooklm/cli/profile.py tests/unit/cli/test_profile.py`
- `uv run mypy src/notebooklm/cli/profile.py`

## Summary by CodeRabbit

* **Improvements**
  * Profile commands now robustly handle missing, unreadable, or corrupt config data and show clearer errors when saving fails.
  * Config and profile directories/files are created with stricter permissions on supported platforms.

* **Bug Fixes**
  * Switching, renaming, and deleting profiles more reliably update and validate the configured default, with safer recovery and warning behavior on partial failures.

* **Tests**
  * Added integration tests covering create/switch/delete/rename flows and related edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/teng-lin/notebooklm-py/pull/427)

### CODE REVIEW

File: src/notebooklm/cli/profile.py [hunk 1]
```diff
@@ -73,6 +73,29 @@ def email_to_profile_name(email: str, *, fallback: str = "account") -> str:
...
+
+
+def _write_config(config_path, data: dict) -> None:
+    """Write global config with private permissions."""
+    config_path.parent.mkdir(parents=True, exist_ok=True, mode=0o700)
+    config_path.write_text(
+        json.dumps(data, indent=2, ensure_ascii=False) + "\n",
+        encoding="utf-8",
+    )
+    config_path.chmod(0o600)
```
[the remark this answers is not in the document]
Comment (teng-lin) [reply src/notebooklm/cli/profile.py]: Addressed in commit 620cbfa: _write_config is now platform-aware, preserves private POSIX permissions on non-Windows, uses default=str JSON serialization, and has regression coverage.

File: src/notebooklm/cli/profile.py [hunk 2]
```diff
@@ -73,6 +73,29 @@ def email_to_profile_name(email: str, *, fallback: str = "account") -> str:
...
+    except (json.JSONDecodeError, OSError):
+        if not suppress_errors:
+            raise
+        return {}
+    return data if isinstance(data, dict) else {}
+
+
+def _write_config(config_path, data: dict) -> None:
+    """Write global config with private permissions."""
+    config_path.parent.mkdir(parents=True, exist_ok=True, mode=0o700)
```
[the remark this answers is not in the document]
Comment (teng-lin) [reply src/notebooklm/cli/profile.py]: Addressed in commit 620cbfa: existing config directories are chmodded to 0o700 on non-Windows, with a regression test for pre-existing loose permissions.
